src/net.c

```diff
@@ -68,24 +68,24 @@ int hyper_send_data(int fd, uint8_t *data, uint32_t len)
 	return 0;
 }
 
-int hyper_send_type(int fd, uint32_t type)
+int hyper_send_type(int fd, uint64_t type)
 {
-	uint8_t buf[4];
+	uint8_t buf[8];
 
-	fprintf(stdout, "hyper send type %d\n", type);
+	fprintf(stdout, "hyper send type %d\n", (int)type);
```

Comment:
move the log out from the function. (and print some log with more info instead if needed.)